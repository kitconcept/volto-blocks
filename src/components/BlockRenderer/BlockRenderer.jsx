import React from 'react';
import PropTypes from 'prop-types';
import { blocks } from '~/config';
import withObjectBrowser from '@plone/volto/components/manage/Sidebar/ObjectBrowser';

/**
 * BlockRenderer container class.
 * @class Form
 * @extends Component
 */
function BlockRenderer(props) {
  if (!props.type) {
    // We could have an empty block, although should be handled somewhere else
    return null;
  }

  const EditBlock = blocks.blocksConfig[props.type].edit;
  const ViewBlock = blocks.blocksConfig[props.type].view;

  if (!props.edit) {
    return <ViewBlock {...props} detached onChangeBlock={() => {}} />;
  }
  if (props.edit) {
    return (
      <EditBlock
        {...props}
        detached
        index={0}
        onSelectBlock={() => {}}
        onFocusPreviousBlock={() => {}}
        onFocusNextBlock={() => {}}
        onAddBlock={() => {}}
        onDeleteBlock={() => {}}
        onMutateBlock={() => {}}
        handleKeyDown={() => {}}
      />
    );
  }
  return '';
}

BlockRenderer.propTypes = {
  edit: PropTypes.bool.isRequired,
  type: PropTypes.string.isRequired,
  block: PropTypes.string.isRequired,
  onChangeBlock: PropTypes.func,
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default withObjectBrowser(BlockRenderer);
