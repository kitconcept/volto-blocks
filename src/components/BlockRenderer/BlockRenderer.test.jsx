import React from 'react';
import renderer from 'react-test-renderer';
import BlockRenderer from './BlockRenderer';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

global.__SERVER__ = true; // eslint-disable-line no-underscore-dangle

test('renders a text block component using the block renderer', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <BlockRenderer
        data={{ url: 'heroimage.jpg' }}
        block="theblockid"
        edit
        type="text"
        selected={false}
        onChangeBlock={() => {}}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
