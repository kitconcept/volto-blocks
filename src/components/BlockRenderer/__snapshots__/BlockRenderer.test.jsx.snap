// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a text block component using the block renderer 1`] = `
<div
  className="mocked-block-edit-image"
/>
`;
